### ISSUE TITLE
Update hostkey handling to avoid race conditions

### ISSUE THREAD

**Issue by atc0005** [atc0005/ansible-role-lxd-testenv#1]
Will probably push forward with the current design, but I'd like to simplify the setup with future revisions. This includes potentially removing the hostkey validation currently performed by the role.

http://hakunin.com/six-ansible-practices

Relevant block:

```
# ~/.ssh/config
# For vagrant virtual machines
Host 192.168.33.* *.myapp.dev
  StrictHostKeyChecking no
  UserKnownHostsFile=/dev/null
  User root
  LogLevel ERROR
```

Using this approach, no key scanning for newly created containers would be needed.

**Comment by atc0005** [comment]
Additional ideas here:

- https://gitlab.erc.monash.edu.au/hpc-team/ansible_cluster_in_a_box/commit/0c2a15c07ae213b1a0968fe29d9dc4f75a31cef9
- https://github.com/ansible/ansible-modules-extras/issues/664#issuecomment-220850472
- http://clouds.freeideas.cz/subdom/clouds/2016/03/23/ansible-built-in-variables/


Re the last entry: 

Noting that the host keys are gathered by Ansible when fact gathering. Since we first gather facts via LXD local connection, we'll have access to the keys without having to scan the keys "remotely"

From the second link:

```yaml
known_hosts:
    path: /etc/ssh/ssh_known_hosts
    name: '{{ansible_fqdn}}'
    key: '{{ansible_fqdn}} ecdsa-sha2-nistp256 {{ansible_ssh_host_key_ecdsa_public}}'
delegate_to: '{{target_host}}'
```

**Comment by atc0005** [comment]
From the second link:

```yaml
known_hosts:
    path: /etc/ssh/ssh_known_hosts
    name: '{{ansible_fqdn}}'
    key: '{{ansible_fqdn}} ecdsa-sha2-nistp256 {{ansible_ssh_host_key_ecdsa_public}}'
delegate_to: '{{target_host}}'
```

**Comment by atc0005** [comment]
This isn't managing the `known_hosts` file, but a similar approach is/was used for managing the `/etc/hosts` file.

```yaml
# TODO: Use blockinfile for this (likely faster)
# Alternatively, a nested loop with two copies of groups['all'] works too
- name: "Update hosts file in container"
  delegate_to: "{{ item[0] }}"
  lineinfile:
    dest: /etc/hosts
    regexp: '.*{{ item[1] }}$'
    line: "{{ hostvars[item[1]].ansible_default_ipv4.address }} {{ item[1] }}"
    state: present
  with_nested:
    - "{{ groups['all'] }}"
    - "{{ groups['all'] }}"
  when:
    - hostvars[item.0].ansible_default_ipv4.address is defined
    - lxd_containers_update_hosts_file
```

This is from some prototype work to convert all plays to targeting `localhost` instead of hosts within the inventory file. The inventory is still used, but the hope is that by going with a localhost target the output would be clearer and any race conditions would be nullified due to looping over a single container at a time. Other benefits would include lower disk I/O by removing attempts to perform actions simultaneously.

All of that said, the above example is one that I'm going to test converting into a `blockinfile` module call with a Jinja2 for loop construct to build the output. The current variation works, but I imagine it is a lot slower than it should be.

**Comment by atc0005** [comment]
This link:

https://github.com/ipspace/NetOpsWorkshop/blob/master/tools/ssh-keys/get-keys.yml

has this example:

```yaml
    - blockinfile:
        dest: "{{known_hosts}}"
        marker: "# {mark} This part managed by Ansible"
        block: |
          {% for h in groups['all'] if hostvars[h].host_keys is defined %}
          {%  if hostvars[h].ansible_port is defined %}
          {%   for key in hostvars[h].host_keys.stdout_lines %}
          [{{ key.partition(" ")[0] }}]:{{ hostvars[h].ansible_port }} {{ key.partition(" ")[2] }}
          {%   endfor %}
          {%  else %}
          {{ hostvars[h].host_keys.stdout }}
          {%  endif %}
          {% endfor %}
      run_once: true
      delegate_to: localhost
```

The idea is to update the syntax to reflect a single run from `localhost` instead of a per-host with delegation to localhost.

**Comment by atc0005** [comment]
The current state of the branch I'm working on once again supports updating the `known_hosts` file. 

For a while I had removed that functionality thinking that it wasn't really needed (and for me I suppose it isn't), but if this role is going to be flexible enough to handle the prior approach *and* my current approach of ignoring host keys for test containers this role will need to have support for both options.

I'm now using the `blockinfile` module for maintaining both the `config` and `known_hosts` file after confirming that the `known_hosts` file supports comments (which allows the 'marker' option for `blockinfile` to function).

**Comment by atc0005** [comment]
Included in #14.
